### PR TITLE
Support vite, remove E2E_ENABLED and build as esm

### PR DESCRIPTION
**Pull request by peetck**
- support vite build by add more condition in `isReactNode`
- build as esm & cjs

### CODE REVIEW

**Review by ReiiYuki — APPROVED** [review]

**Review by ReiiYuki — CHANGES_REQUESTED** [review]
After internal discussion, please allow e2e env

File: package.json [hunk 1]
```diff
@@ -17,6 +18,17 @@
     "e2e",
     "data-qa"
   ],
+  "main": "./index.js",
+  "types": "./index.d.ts",
+  "exports": {
+    ".": {
+      "types": {
+        "default": "./index.d.ts"
+      },
```
Comment (nitpum) [inline package.json]: types exports no need to split variants
```suggestion
      "types": "./index.d.ts",
```

Comment (nitpum) [reply package.json]: Or something like this instead. [ref](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing)
```ts
            "import": {
                // Where TypeScript will look.
                "types": "./types/esm/index.d.ts",
                // Where Node.js will look.
                "default": "./esm/index.js"
            },
            // Entry-point for `require("my-package") in CJS
            "require": {
                // Where TypeScript will look.
                "types": "./types/commonjs/index.d.cts",
                // Where Node.js will look.
                "default": "./commonjs/index.cjs"
            },
```

**Review by nitpum — CHANGES_REQUESTED** [review]

**Comment by ppipee** [comment]
LGTM

**Pull request merged**